### PR TITLE
Fixes #17570: Trigger promises generation when promoting a node to relay

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/17570

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Are you sure it needs a reload ? i don't think because rule creation already reloads

**Comment by ElaadF** [comment]
I don't see where the reload is performed, no eventlog indicates this and it seems that in https://github.com/Normation/rudder/blob/master/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPRuleRepository.scala#L186 there is no reload at this point

File: scale-out-relay/src/main/scala/com/normation/plugins/scaleoutrelay/ScaleOutRelayService.scala [hunk 1]
```diff
@@ -59,6 +61,7 @@ class ScaleOutRelayService(
                     } else {
                       nodeInfos.succeed
                     }
+      _ <- clearCacheService.clearNodeConfigurationCache(storeEvent = true, actor)
```
Comment (VinceMacBuche) [inline scale-out-relay/src/main/scala/com/normation/plugins/scaleoutrelay/ScaleOutRelayService.scala:64]: In fact you should trigger a deployment not delete the whole configurationCache.

We used to call deploy api that only starts a deployment:

```
      asyncDeploymentAgent ! AutomaticStartDeployment(modId, actor)
```

with 
```
  , asyncDeploymentAgent : AsyncDeploymentActor
```

you can put it in the yield section

**Comment by ElaadF** [comment]
Commit modified

**Pull request merged**